x-pack/plugins/apm/public/components/app/service_overview/index.tsx

```diff
@@ -34,17 +36,29 @@ import { useTimeRange } from '../../../hooks/use_time_range';
 export const chartHeight = 288;
 
 export function ServiceOverview() {
-  const { agentName, serviceName } = useApmServiceContext();
+  const { agentName, serviceName, transactionType } = useApmServiceContext();
   const {
     query,
-    query: { environment, kuery, rangeFrom, rangeTo },
+    query: {
+      environment,
+      kuery,
+      rangeFrom,
+      rangeTo,
+      transactionType: transactionTypeFromUrl,
+    },
   } = useApmParams('/services/:serviceName/overview');
   const { fallbackToTransactions } = useFallbackToTransactionsFetcher({
     kuery,
   });
-
   const { start, end } = useTimeRange({ rangeFrom, rangeTo });
 
+  const history = useHistory();
+
+  // redirect to first transaction type
+  if (!transactionTypeFromUrl && transactionType) {
+    replace(history, { query: { transactionType } });
+  }
```

Comment:
Was this in there before?

Reply:
I'm curious if we can make the `transactionType` query parameter required for the transaction overview. What happens if you make that change, what type errors do you see?

Reply:
> Was this in there before?
No, it wasn't. But I think it makes sense to add it here too since it's required.

Reply:
> I'm curious if we can make the `transactionType` query parameter required for the transaction overview. What happens if you make that change, what type errors do you see?

@dgieselaar does it make sense to make it required? I mean there's no default value to set when it is not defined in the URL, right? I must wait the response from the API call to set Request/Page Load or the first item available in the list.

Reply:
That might not be the case. In most cases where we link to the transaction overview or detail page we already have a transaction type. Then, I'm hoping we can use the transaction type from the query parameter to start loading data rather than waiting on a blocking request which will likely return the same result.

Reply:
IMO this is another issue, which could be handle on a separate PR, this one supposes to fix the problem with the alert creation.

Reply:
I can see that, but I'm not sure if the redirect should be there then either.

Reply:
Why is that?

Reply:
Because I would wager that the issue is not the fact that there is no transaction type in the URL, but that it's because the correct transaction type is _displayed_ in the alerting flyout, but somehow not passed to the API. If the issue was the URL parameter, I'd assume there would be no transaction type in the flyout either, but maybe I'm missing something?

Reply:
No, you're correct. The transaction type is available in the flyout because it reads it from [useServiceTransactionTypesFetcher](https://github.com/cauemarcondes/kibana/blob/apm-alert-transactionType-error/x-pack/plugins/apm/public/components/alerting/transaction_duration_alert_trigger/index.tsx#L80:L81) but the `alertParams` doesn't have it defined by default.

Reply:
And the AlertFlyout reads it from the URL https://github.com/cauemarcondes/kibana/blob/apm-alert-transactionType-error/x-pack/plugins/apm/public/components/alerting/alerting_flyout/index.tsx#L41:L42, which is not defined when we navigate to the service overview page